nixos/modules/services/monitoring/prometheus/alertmanager-irc-relay.nix

```
@@ -0,0 +1,96 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.prometheus.alertmanagerIrcRelay;
+
+  configFormat = pkgs.formats.yaml {};
+  configFile = configFormat.generate "alertmanager-irc-relay.yml" cfg.settings;
+in {
+  options.services.prometheus.alertmanagerIrcRelay = {
+    enable = mkEnableOption "Alertmanager IRC Relay";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.alertmanager-irc-relay;
+      defaultText = literalExpression "pkgs.alertmanager-irc-relay";
+      description = "Alertmanager IRC Relay package to use.";
+    };
+
+    settings = mkOption {
+      type = types.attrsOf configFormat.type;
```

```suggestion
      type = configFormat.type;
```

That should be sufficient, since YAML already allows nested attribute sets.